src/pages/experiments/[accession_code]/index.js

```diff
@@ -1,29 +1,33 @@
 import { useRouter } from 'next/router'
 import { useEffect } from 'react'
-import { useExperiments } from 'hooks/useExperiments'
+import { api } from 'api'
 import formatURLString from 'helpers/formatURLString'
 
 // uses the query parameter 'accession code' to redirect to an experiment page with its title
-export const AccessionCode = () => {
-  const {
-    query: { accession_code: accessionCode },
-    isReady,
-    push
-  } = useRouter()
-
-  const { experiment, getExperiment } = useExperiments()
+export const AccessionCode = ({ experiment }) => {
+  const { push } = useRouter()
 
   useEffect(() => {
-    if (isReady) {
-      getExperiment(accessionCode)
+    push(
+      `/experiments/${experiment.accession_code}/${formatURLString(
+        experiment.title
+      )}`
+    )
+  }, [])
+}
 
-      if (experiment) {
-        push(
-          `/experiments/${accessionCode}/${formatURLString(experiment.title)}`
-        )
-      }
+export const getServerSideProps = async ({ query }) => {
+  const response = await api.experiments.get(query.accession_code)
+
+  if (!response) {
+    return {
+      notFound: true
     }
-  }, [isReady, experiment])
+  }
+
+  return {
+    props: { experiment: response }
+  }
```

Comment:
Wouldn't it make more sense to use a redirect here?

https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props#redirect

Then the page component could be pretty much empty.

Reply:
Great catch, thank you 👍 !